src/CounterEIP712.sol

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.18;
+
+import "openzeppelin-contracts/contracts/access/Ownable.sol";
+
+import {IERC1271} from "./interface/IERC1271.sol";
+
+/**
+ * @title CounterEIP712
+ * @dev A smart contract that allows secure updates of a number using the EIP-712 standard.
+ * The contract is owned by an address that has the authority to verify and set the number.
+ */
+contract CounterEIP712 is Ownable {
+    // EIP-712 domain separator
+    bytes32 private DOMAIN_SEPARATOR;
+
+    // Number, which can get changed with setNumber
+    uint256 public number;
+
+    /**
+     * @dev Constructor to initialize the contract owner and set the EIP-712 domain separator.
+     * @param _addr The address of a contract to be used as a domain separator.
+     */
+    constructor(address _addr) Ownable(_addr) {
+        require(_addr.code.length > 0, "Not a contract");
+        DOMAIN_SEPARATOR = keccak256(
+            abi.encode(
+                keccak256(
```

Comment:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/94697be8a3f0dfcd95dfb13ffbd39b5973f5c65d/contracts/utils/cryptography/EIP712.sol#L37

make it a constant